mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb

```diff
@@ -156,6 +156,124 @@ func TestAccContainerNodePool_noName(t *testing.T) {
 	})
 }
 
+func TestAccContainerNodePool_withNoSpecifiedLoggingVariant(t *testing.T) {
+	t.Parallel()
+
+	cluster := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	nodePool := fmt.Sprintf("tf-test-nodepool-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccContainerNodePool_basic(cluster, nodePool),
+				Check: resource.TestCheckResourceAttr("google_container_node_pool.np",
+					"node_config.0.logging_variant",
+					"DEFAULT"),
+			},
+			resource.TestStep{
+				ResourceName:      "google_container_node_pool.np",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccContainerNodePool_withDefaultLoggingVariant(t *testing.T) {
+	t.Parallel()
+
+	cluster := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	nodePool := fmt.Sprintf("tf-test-nodepool-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccContainerNodePool_withLoggingVariant(cluster, nodePool, "DEFAULT"),
+				Check: resource.TestCheckResourceAttr("google_container_node_pool.with_logging_variant",
+					"node_config.0.logging_variant",
+					"DEFAULT"),
+			},
+			resource.TestStep{
+				ResourceName:      "google_container_node_pool.with_logging_variant",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccContainerNodePool_withMaxThroughputLoggingVariant(t *testing.T) {
+	t.Parallel()
+
+	cluster := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	nodePool := fmt.Sprintf("tf-test-nodepool-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccContainerNodePool_withLoggingVariant(cluster, nodePool, "MAX_THROUGHPUT"),
+				Check: resource.TestCheckResourceAttr("google_container_node_pool.with_logging_variant",
```

Comment:
generally we don't include explicit checks for values because the values will usually be automatically confirmed by the test framework, which checks to ensure that there are no diffs after the config is applied.

```suggestion
```

Reply:
Ack. Do we need to test if the GKE API effectively took the intended action by fetching a cluster independently of Terraform state, or is it enough to just follow the steps recommended in https://googlecloudplatform.github.io/magic-modules/docs/handwritten/#testing (which I followed in the latest changes pushed to github)?